packages/typescript-estree/src/tsconfig-parser.ts

```diff
@@ -62,14 +82,42 @@ function diagnosticReporter(diagnostic: ts.Diagnostic): void {
   );
 }
 
-const noopFileWatcher = { close: (): void => {} };
-
 function getTsconfigPath(tsconfigPath: string, extra: Extra): string {
   return path.isAbsolute(tsconfigPath)
     ? tsconfigPath
     : path.join(extra.tsconfigRootDir || process.cwd(), tsconfigPath);
 }
 
+interface Watcher {
+  close(): void;
+  on(evt: 'add', listener: (file: string) => void): void;
+  on(evt: 'change', listener: (file: string) => void): void;
+}
+/**
+ * Watches a file or directory for changes
+ */
+function watch(
+  path: string,
+  options: chokidar.WatchOptions,
+  extra: Extra,
+): Watcher {
+  // an escape hatch to disable the file watchers as they can take a bit to initialise in some cases
+  // this also supports an env variable so it's easy to switch on/off from the CLI
+  if (process.env.PARSER_NO_WATCH === 'true' || extra.noWatch === true) {
+    return {
+      close: (): void => {},
+      on: (): void => {},
+    };
+  }
+
+  return chokidar.watch(path, {
```

Comment:
Have you considered adding `usePolling: false` (at least when `TSC_WATCHFILE` is provided)? TypeScript has some special logic there (see https://github.com/microsoft/TypeScript/issues/31048#issuecomment-495659901), but it's probably to complex to copy it. Also see https://github.com/microsoft/TypeScript/pull/21243#issue-163557829 and https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/256#issuecomment-485245806 for more context

Reply:
I didn't set it explicitly, because false is the default value for the option in chokidar.

Reply:
Oh right, sorry, I've missed that somehow